cloudinit/sources/DataSourceAzure.py

```diff
@@ -1431,7 +1471,7 @@ def get_metadata_from_imds(fallback_nic, retries):
 @azure_ds_telemetry_reporter
 def _get_metadata_from_imds(retries):
 
-    url = IMDS_URL + "instance?api-version=2017-12-01"
+    url = IMDS_URL + "instance?api-version=2019-06-01"
```

Comment:
is azure-stack guaranteed to have this api version?

Reply:
Whether Azure Stack doesn't have that version, or no IMDS at all, a failure is a failure here. The situation is the same. The only consideration is are we causing an issue for existing Azure Stack implementations that *do* have IMDS but do *not* have api-version 2019-06-01. I believe that this is not the case or a concern. @anhvoms, let me know if you disagree with that statement.

Reply:
There is no IMDS support in Azure Stack currently. Once they do I would expect them to pick up the latest version

Reply:
Great thanks for the clarification, @anhvoms. So it sounds like this won't be a breaking change for Azure Stack then.